sycl/include/sycl/info/device_traits.def

```diff
@@ -237,6 +237,9 @@ __SYCL_PARAM_TRAITS_SPEC(device, ext_oneapi_max_work_groups_3d, id<3>,
 __SYCL_PARAM_TRAITS_SPEC(device, ext_oneapi_max_global_work_groups, size_t, __SYCL_TRAIT_HANDLED_IN_RT)
 __SYCL_PARAM_TRAITS_SPEC(device, ext_oneapi_cuda_cluster_group, bool, __SYCL_TRAIT_HANDLED_IN_RT)
 
+__SYCL_PARAM_TRAITS_SPEC(device, ext_intel_device_info_luid, detail::luid_type,
+                         UR_DEVICE_INFO_LUID)
```

Comment:
Existing code isn't strict in following it, but `__SYCL_TRAIT_HANDLED_IN_RT` would be "more" correct as we're doing some exception handling in the implementation and have an explicit `CASE`.

Reply:
Added.

Just to check my understanding, so if the exception handling was not needed, then you could keep the `UR_DEVICE_INFO_LUID` here and it would automatically hook up UR and the SYCL RT without an explicit case.

Reply:
Yes, there are lots of traits that don't have explicit c++ handling and just rely on this mapping (and a separate type mapping between this UR_<something> to UR's return type). Conversion between UR's return type and SYCL return type (defined here) has to be either implicit or one of the pre-defined supported ones (like maybe ur_device_handle_t -> sycl::device, IIRC, or something like that).